presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkDiskBasedBroadcastDependency.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark;
+
+import com.facebook.airlift.log.Logger;
+import com.facebook.presto.spark.classloader_interface.PrestoSparkDiskPage;
+import com.facebook.presto.spi.storage.TempDataOperationContext;
+import com.facebook.presto.spi.storage.TempStorage;
+import io.airlift.units.DataSize;
+import org.apache.spark.SparkEnv;
+import org.apache.spark.SparkException;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.broadcast.Broadcast;
+import org.apache.spark.storage.BroadcastBlockId;
+import scala.Tuple2;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.List;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.Collectors;
+
+import static com.facebook.presto.ExceededMemoryLimitException.exceededLocalBroadcastMemoryLimit;
+import static com.facebook.presto.spark.util.PrestoSparkUtils.computeNextTimeout;
+import static io.airlift.units.DataSize.succinctBytes;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.stream.Collectors.toList;
+
+public class PrestoSparkDiskBasedBroadcastDependency
+        implements PrestoSparkBroadcastDependency<PrestoSparkDiskPage>
+{
+    private static final Logger log = Logger.get(PrestoSparkDiskBasedBroadcastDependency.class);
+
+    // The value starts from 100000 to avoid any collision with broadcastId generated
+    // by spark internally which starts from 0
+    private static final AtomicLong nextBroadcastId = new AtomicLong(100000);
```

Comment:
Let's make the `broadcastId ` a String and use a UUID, that is simpler and more reliable. After you use a local temporary file as a cache the requirement for the ID to be a long will no longer be there